packages/react/focus-scope/src/FocusScope.stories.tsx

```diff
@@ -264,3 +294,162 @@ export const WithOptions = () => {
+}) {
+  return (
+    <span
+      style={{
+        color: color,
+        position: 'absolute',
+        top: 'calc(100% + 0.25em)',
+        fontSize: '12px',
+        fontFamily: 'sans-serif',
+        left: 0,
+        opacity: on ? 1 : 0,
+        transition: on ? undefined : 'opacity  1s ease-in-out',
+      }}
+    >
+      {children}
+    </span>
+  );
+}
+
+function useSelfDestructiveToggleState(
+  initialState: boolean,
+  opts: { timeout?: number } = {}
+): [boolean, React.Dispatch<React.SetStateAction<boolean>>] {
+  const { timeout = 500 } = opts;
+  const [on, setOn] = React.useState(initialState);
+  const savedTimeoutValue = React.useRef(timeout);
+  React.useEffect(() => {
+    savedTimeoutValue.current = timeout;
+  });
+  React.useEffect(() => {
+    const timeout = savedTimeoutValue.current;
+    let mounted = true;
+    if (on) {
+      const timeoutId = window.setTimeout(() => {
+        if (mounted) {
+          setOn(false);
+        }
+      }, timeout);
+      return () => {
+        mounted = false;
+        window.clearTimeout(timeoutId);
+      };
+    }
+    return () => {
+      mounted = false;
+    };
+  }, [on]);
+
+  return [on, setOn];
+}
+
+const RadioGroupContext = React.createContext<{
+  name?: string;
+  checked?: string;
+  getChangeHandler?(value: string): (event: React.ChangeEvent<HTMLInputElement>) => void;
+}>({});
+
+function RadioGroup({
+  children,
+  name,
+  checked,
+  onChange,
+  ...props
+}: Omit<React.ComponentProps<'fieldset'>, 'onChange'> & {
+  name: string;
+  checked: string;
+  onChange(value: string): void;
+}) {
+  function getChangeHandler(value: string) {
+    return function handleChange(event: React.ChangeEvent<HTMLInputElement>) {
+      if (event.target.checked) {
+        onChange(value);
+      }
+    };
+  }
+
+  return (
+    <Fieldset {...props}>
+      <RadioGroupContext.Provider value={{ name, checked, getChangeHandler }}>
+        {children}
+      </RadioGroupContext.Provider>
+    </Fieldset>
+  );
+}
+
+function Fieldset({ ...props }: React.ComponentProps<'fieldset'>) {
+  return <fieldset {...props} />;
+}
+
+function Legend({ ...props }: React.ComponentProps<'legend'>) {
+  return <legend {...props} />;
+}
+
+function Radio({ ...props }: Omit<React.ComponentProps<'input'>, 'type'> & { value: string }) {
+  const { name, checked, getChangeHandler } = React.useContext(RadioGroupContext);
+  const handleChange = props.onChange || getChangeHandler?.(props.value);
+  return (
+    <CheckedInput
+      name={name}
+      checked={checked != null ? checked === props.value : undefined}
+      onChange={handleChange}
+      {...props}
+      type="radio"
+    />
+  );
+}
+
+function CheckedInput({ children, ...props }: React.ComponentProps<'input'>) {
+  return (
+    <label style={{ display: 'flex', marginTop: '0.25em' }}>
+      <input {...props} />
+      <span style={{ display: 'inline-block', marginLeft: '0.25em' }}>{children}</span>
+    </label>
+  );
+}
```

Comment:
Not sure if you meant to leave this all here or if it was just for your own testing but I feel like we should remove this as it detracts massively from the story itself. If it's to check that other focus/blur events do happen properly I feel like an inline `onFocus`, `onBlur` handler on the things we want to check with a console.log is good enough and doesn't detract much from the story.

Reply:
Yeah we can totally remove it. Let's keep it until we settle on a proper fix for quick testing, but I'm glad to pull it out before we merge.